src/screens/Home/index.js

```diff
@@ -44,16 +44,23 @@ class HomeScreen extends Component {
     }
 
     showToast = (prevPropsHomeDocuments, prevStateShowToast) => {
+        const { homeDocuments } = this.props;
+        const areDocumentsEqual = JSON.stringify(homeDocuments) === JSON.stringify(prevPropsHomeDocuments);
```

Comment:
Try having the JSON.stringify comparison happen in the Document screen - this way the JSON.stringify() isn't having to go through a potentially long array of objects. It'd be better to have just the edited document (this.state.document) compared with the original existing document (this.props.document).